treehasher.go

```diff
@@ -76,5 +75,9 @@ func (th *treeHasher) pathSize() int {
 }
 
 func (th *treeHasher) placeholder() []byte {
-	return bytes.Repeat([]byte{0}, th.pathSize())
+	size := th.pathSize()
+	if size > len(th.ph) {
+		th.ph = bytes.Repeat([]byte{0}, size)
+	}
+	return th.ph[:size]
```

Comment:
`th.pathSize()` will be a constant for any `treeHasher` instance, so semantically this should be something like

```suggestion
	if len(th.ph) == 0 {
		size := th.pathSize()
		th.ph = bytes.Repeat([]byte{0}, size)
	}
	return th.ph
```

(functionally equivalent)

Reply:
@adlerjohn Ah, good point about the `th.pathSize()` is constant. Then I think why don't we just create it once when creating new `treeHasher`? I proposed this change because I think `th.pathSize()` can be changed.

Reply:
> Then I think why don't we just create it once when creating new `treeHasher`?

That's an excellent suggestion!